tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -285,7 +285,62 @@ object SQLPlanParser extends Logging {
         case _ => // NO OP
       }
     }
-    parsedExpressions.toArray
+    parsedExpressions.distinct.toArray
+  }
+
+  // This parser is used for SortAggregateExec, HashAggregateExec and ObjectHashAggregateExec
+  def parseAggregateExpressions(exprStr: String): Array[String] = {
+    val parsedExpressions = ArrayBuffer[String]()
+    // (key=[num#83], functions=[partial_collect_list(letter#84, 0, 0), partial_count(letter#84)])
+    val pattern = """functions=\[([\w#, +*\\\-\.<>=\`\(\)]+\])""".r
+    val aggregatesString = pattern.findFirstMatchIn(exprStr)
+    // This is to split multiple column names in AggregateExec. Each column will be aggregating
+    // based on the aggregate function. Here "partial_" is removed and only function name is
+    // preserved. Below regex will first remove the "functions=" from the string followed by
+    // removing "partial_". That string is split which produces an array containing
+    // column names. Finally we remove the parentheses from the beginning and end to get only
+    // the expressions. Result will be as below.
+    // paranRemoved = Array(collect_list(letter#84, 0, 0),, count(letter#84))
+    if (aggregatesString.isDefined) {
+      val paranRemoved = aggregatesString.get.toString.replaceAll("functions=", "").
+          replaceAll("partial_", "").split("(?<=\\),)").map(_.trim).
+          map(_.replaceAll("""^\[+""", "").replaceAll("""\]+$""", ""))
+      val functionPattern = """(\w+)\(.*\)""".r
+      paranRemoved.foreach { case expr =>
+        val functionName = getFunctionName(functionPattern, expr)
+        functionName match {
+          case Some(func) => parsedExpressions += func
+          case _ => // NO OP
+        }
+      }
+    }
+    parsedExpressions.distinct.toArray
+  }
+
+  def parseSortExpressions(exprStr: String): Array[String] = {
+    val parsedExpressions = ArrayBuffer[String]()
+    // Sort [round(num#126, 0) ASC NULLS FIRST, letter#127 DESC NULLS LAST], true, 0
```

Comment:
in the plugin when this is parsed, is the round here really part of the sort of is it broken out into Project?  I guess it doesn't hurt for us to check here as well just wondering if it gets checked twice.

Reply:
The query I ran had `round` expr in `Sort` itself. So in the plan it shows up in SortExec. And if we have separate round in `select` then it will show up in Project Exec.
```
val df1 = Seq((1.7, "a"), (1.6, "aa"), (1.1, "b"), (2.5, "a"),(2.2, "b"), (3.2, "a"), (3.6, "c")).toDF("num", "letter")
df1.select(col("num"), col("letter")).sort(round(col("num")), col("letter").desc).explain

== Physical Plan ==
GpuColumnarToRow false
+- GpuSort [gpuround(num#7, 0, DoubleType) ASC NULLS FIRST, letter#8 DESC NULLS LAST], true, com.nvidia.spark.rapids.OutOfCoreSort$@1e9cc446
   +- GpuShuffleCoalesce 2147483647
      +- GpuColumnarExchange gpurangepartitioning(gpuround(num#7, 0, DoubleType) ASC NULLS FIRST, letter#8 DESC NULLS LAST, 200), ENSURE_REQUIREMENTS, [id=#33]
         +- GpuRowToColumnar targetsize(2147483647)
            +- LocalTableScan [num#7, letter#8]
```